### PR TITLE
FIRMessagingCheckinServiceTest: use OCMock to stub NSURLSession

### PR DESCRIPTION
**Pull request by maksymmalyhin**

### CODE REVIEW

File: FirebaseMessaging.podspec [hunk 1]
```diff
@@ -36,10 +36,14 @@ device, and it is completely free.
     'Firebase/InstanceID/Public/*.h',
     'FirebaseInstallations/Source/Library/Private/*.h',
   ]
-  s.requires_arc = [
-    base_dir + 'Sources/*.m',
-    base_dir + 'Sources/Token/*.m'
-  ]
+
+  # TODO: Find out why `requires_arc` was required.
```
Comment (paulb777) [inline FirebaseMessaging.podspec:40]: I suspect, it's left over from the when the Protobuf dependency required some files to be built without ARC

Comment (maksymmalyhin) [reply FirebaseMessaging.podspec:40]: If so, looks like safe to be removed.

Comment (paulb777) [reply FirebaseMessaging.podspec:40]: Agreed

**Comment by maksymmalyhin** [comment]
Other test failures seems to be inherited from the parent branch and not relevant to the PR changes.

File: FirebaseMessaging.podspec [hunk 2]
```diff
@@ -36,10 +36,7 @@ device, and it is completely free.
     'Firebase/InstanceID/Public/*.h',
     'FirebaseInstallations/Source/Library/Private/*.h',
   ]
-  s.requires_arc = [
```
Comment (chliangGoogle) [inline FirebaseMessaging.podspec:39]: curious: this is not needed?

**Review by chliangGoogle — APPROVED** [review]
Awesome. thank you for doing this Maksym!

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinServiceTest.m [hunk 3]
```diff
@@ -47,23 +40,29 @@ @implementation FIRMessagingCheckinServiceTest
 
 - (void)setUp {
   [super setUp];
+
+  // Stub NSURLSession constructor before instantiating FIRMessagingCheckinService to inject
+  // URLSessionMock.
+  // TODO: inject NSURLSession instance via an initializer.
+  self.URLSessionMock = OCMClassMock([NSURLSession class]);
+  OCMStub(ClassMethod([self.URLSessionMock sessionWithConfiguration:[OCMArg any]]))
```
Comment (chliangGoogle) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinServiceTest.m:48]: Nice! looks like I don't even have to expose the local session object.

File: FirebaseMessaging.podspec (hunk 2 above)
Comment (maksymmalyhin) [reply FirebaseMessaging.podspec:39]: If the [reason mentioned by Paul](https://github.com/firebase/firebase-ios-sdk/pull/7617#discussion_r583771492) is the only one, than I think we don't need it any more.

**Pull request merged**